README.md

```diff
@@ -36,40 +48,80 @@ Quick start
         authorization_grant_type=Application.GRANT_CLIENT_CREDENTIALS,
     }
 
-4. On the client side, put client and secret from receiver side in .env and instantiate client via library calls, and use it for all api calls.
+4. On the client side, create oauth2_client.Application model instance, and
+populate it. You can get help by calling `python manage.py oauth2client_app -h`.
 
+5. Instantiate the client via library calls, and use it for all api calls.
 
-Test
-----
 
-- `pip install tox`
-- get your test postgres instnce running, see points below
-- (optional, if you want custom settings) export variables for tox
-- `tox`
+Tests and Development
+---------------------
+Tests use PostgreSQL. The setup is automated with docker-compose, that is
+installed locally with `make docker_setup` command.  
+Project requirements live inside `requirements/tox_requirements{27|37}.txt`.  
+Test settings live inside `test_settings.py`.
 
-### More info
-As it is Django app without Django project, custom test runnner created: `test_runnner.py`
-Without tox, to run tests, install requirements from needed python version (see `tox_requirements....txt`) and then `python test_runnner.py`. Inspiration: https://stackoverflow.com/questions/3841725/how-to-launch-tests-for-django-reusable-app
 
-Tests use PostgresDb. Install postgres, create db, put its settings into test_runnner.py
+#### Development Requirements
+- python3.7 or higher. Python2.7 is supported but discouraged.
+- [virtualenv](https://virtualenv.pypa.io/en/latest/)
+- [docker](https://docs.docker.com/install/)
 
-#### Codestyle
+#### Automated Tox Run
+- tox runs tests for both python2 and python3
+- in system-wide python: `pip install tox`
+- `cd PROJECT_ROOT` - root of this project
+- `mkdir -p reports/diff_coverage`
+- `make docker_setup`  # set up a venv for docker-compose
+- `make start && sleep 5`  # start a dockerized PostgreSQL instance
+- `tox`
+
+#### Running Tests in IDE (Tested With Pycharm)
+- `make setup`  # create a venv inside a `.venv` dir and installs dependencies,
+this is hardcoded to use python3
+- `make docker_setup`  # set up a venv for docker-compose
+- `make start && sleep 5`  # start a dockerized PostgreSQL instance
+- `python test_manage.py migrate`  # apply migrations
+- point your IDE at `.venv` as the python installation to be used
+- tests should work in the IDE
+
+#### Running Tests in Django-style
+As it is a Django app without a Django project, custom `manage.py` created:
+`test_manage.py`. It is a cut-down framework to provide sufficient Django
+context for testing and uses a test settings file, `test_settings.py`.  
+To run tests as if it was a Django project:
+- `make setup`  # create a venv inside a `.venv` dir and installs dependencies,
+this is hardcoded to use python3
+- `make docker_setup`  # set up a venv for docker-compose
+- `make start && sleep 5`  # start a dockerized PostgreSQL instance
+- `source .venv/bin/activate`
+- `python test_manage.py test`
```

Comment:
Just doublechecking: are we still able to run tests when this code is inside another Django project?

Reply:
Hi @auraz , I don't fully understand your question so please clarify, ideally providing a few lines of shell (can be pseudocode) that you would like to run and have Client's tests triggered as a result. 

Reason of my confusion: 
- Client is indented to be installed as a regular python dependency as described in the Readme: https://raw.githubusercontent.com/Livit/Labster.OAuth2Client/master/README.md 
- I've never heard of a software project running tests of its dependencies upon triggering parent project's build. Installation via pip is a different story and my understanding is that the Client currently doesn't support that installation method.

Reply:
@fredmajor Sorry for confusion, probably it is mine misunderstanding.

Btw, we are installing it via pip in other projects https://github.com/Livit/Labster.OAuth2Client/pull/9/files#diff-04c6e90faac2675aa89e2176d2eec7d8R29

Reply:
OK I see, my bad this time :) I quickly googled and we could somehow hook up pip's test run to our tests, but that would be out of scope of this PR. 

This is because current test architecture depends on the user having a running postgres instance. Currently this is automated with docker compose, what is fine for devs interested in working with this project, but sounds like a too heavy requirement for a casual user of this lib, who wants to just run tests to make sure the lib installs fine (at least this is my opinion)

Although the way this project uses the database is rather trivial, if having the test run "lighter" is a requirement, it should be very easy to migrate to sqllite. That would reduce external test dependency on docker/postgres.

Reply:
It isn't great that testing requires a functioning database service at all. Could we perhaps mock it for testing? This suggestion could be implemented as a later ticket if testing currenty suffices.